lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/NotificationHelper.kt

```diff
@@ -0,0 +1,364 @@
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+@Suppress("unused")
+class NotificationHelper {
+    companion object {
+        private const val TAG = "NotificationHelper"
+        private var defaultClickAction: String? = null
+
+        fun getNotificationManager(context: Context): NotificationManager? {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                val notificationManager =
+                    context.getSystemService(Context.NOTIFICATION_SERVICE)
+                            as NotificationManager
+                if (notificationManager.areNotificationsEnabled()) {
+                    return notificationManager
+                }
+            }
+            return null
+        }
+
+        @SuppressLint("NewApi")
+        fun createNotificationChannelGroup(
+            context: Context,
+            groupId: String,
+            groupName: String,
+            groupDescription: String,
+        ) {
+            getNotificationManager(context)?.also { manager ->
+                val channelGroup = NotificationChannelGroup(
+                    groupId,
+                    groupName,
+                )
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                    channelGroup.description = groupDescription
+                }
+
+                manager.createNotificationChannelGroup(channelGroup)
+            }
+        }
+
+        @SuppressLint("NewApi")
+        fun createNotificationChannel(
+            context: Context,
+            channelId: String,
+            channelName: String,
+            channelDescription: String,
+            groupId: String,
+            importance: Int = NotificationManager.IMPORTANCE_DEFAULT,
+        ) {
+            getNotificationManager(context)?.also { manager ->
+                val applicationId = context.applicationContext.packageName
+                val notificationChannel = NotificationChannel(
+                    "${applicationId}.${channelId}",
+                    channelName,
+                    importance
+                ).apply {
+                    description = channelDescription
+                    group = groupId
+                }
+
+                manager.createNotificationChannel(notificationChannel)
+            }
+        }
+
+        @SuppressLint("NewApi")
+        fun registerNotificationChannels(context: Context, defaultClickAction: String? = null) {
+            this.defaultClickAction = defaultClickAction
+
+            getNotificationManager(context)?.also { manager ->
+                createNotificationChannelGroups(context, manager)
+                createNotificationChannels(context, manager)
+                Log.d(TAG, "Registered notification channels")
+            }
+        }
+
+        @RequiresApi(Build.VERSION_CODES.O)
+        private fun createNotificationChannels(
```

Comment:
Reviewing this again made me aware of parts we could improve. I really want to refactor this function and make it more modular :)

Which poses a question: How can we ensure changes made in this plugin are also reflected in the GL notification plugin, and vice versa?

Reply:
I agree, there is overlap. I think we agreed in the dev call to keep it separate to begin with, then work together a more modular approach. I would suggest a separate repo for the core plugin components

Reply:
Let's follow our decision to first implement a workable solution then we will better understand on how to proceed.